api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/controllers/api/ApiCatalogController.java

```diff
@@ -64,6 +68,15 @@ public ApiCatalogController(CachedProductFamilyService cachedProductFamilyServic
         this.cachedApiDocService = cachedApiDocService;
     }
 
+    @GetMapping(value = "/oidc/provider", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<String>> getOidcProvider() {
+        if (oidcProviderCache.get() == null) {
+            oidcProviderCache.set(OidcUtils.getOidcProvider());
+        }
+
+        return new ResponseEntity<>(oidcProviderCache.get(), HttpStatus.OK);
```

Comment:
it may be better to return 204 when there no OIDC provider

Reply:
changed